main.go

```diff
@@ -1184,98 +1267,74 @@ func sign(num float64) float64 {
 	return 1.0
 }
 
-func distBetweenAngles(endDeg, startDeg, originalDist float64) float64 {
-	totalDeg := endDeg - startDeg
-	if sign(totalDeg) != sign(originalDist) && !rdkutils.Float64AlmostEqual(0, totalDeg, 10) {
-		totalDeg += sign(originalDist) * 360
-	}
-	turns := (int(originalDist) / 360.0) % 360
-	return totalDeg + float64(turns*360)
+func distBetweenAngles(endRad, startRad, originalDist float64) float64 {
+	turns := int(originalDist) / 360 % 360
+	total := rdkutils.RadToDeg((math.Sin(endRad) - math.Sin(startRad))) + float64(turns)*360
+	return total
 }
 
-// timeEst must be absolute value
-func sampleLinearVel(odometry movementsensor.MovementSensor, goalVel, timeEst float64, data *os.File, testType string, startPos float64) float64 {
-	maxSpeed := -0.000000001
+func sampleEverything(ctx context.Context, odometry movementsensor.MovementSensor, m *motor.Motor, goalLinVel, goalAngVel, timeEst float64, data *os.File, testType string, cancel func()) (float64, float64) {
+	bestLin, bestAng := 0.0, 0.0
+	maxLin, maxAng := 0.0, 0.0
 	start := time.Now()
-	for {
-		linVel, err := odometry.LinearVelocity(context.Background(), nil)
-		if err != nil {
-			logger.Error(err)
-			return -1
+	ticker := time.NewTicker(50 * time.Millisecond)
```

Comment:
ah different functions have different tickers. should be fine, so can maybe ignore the other comment about making these tickers constants. up to you if you want to